### PR TITLE
kt: Return a common error type

### PR DESCRIPTION
**Pull request by lmb**
This makes it easier to distinguish KT errors at runtime.
cc @pavel-odintsov

### CODE REVIEW

File: kt/kt.go [hunk 1]
```diff
@@ -17,6 +16,18 @@ import (
...
+// Error is returned by all functions in this package.
+type Error struct {
+	// Error returned by KT
+	Message string
+	// HTTP status code, if any (0 otherwise)
+	Code int
+}
+
+func (e *Error) Error() string {
+	return "kt: " + e.Message
```
Comment (pavel-odintsov) [inline kt/kt.go]: What about code like this to expose more information about return code:
```
if e.Code == 0 {
    return "kt: " + e.Message 
} else {
   return fmt.Sprintf("kt: %s code: %d", e.Message, e.Code)
}
```

File: kt/kt.go [hunk 2]
```diff
@@ -17,6 +16,18 @@ import (
 
 const DEFAULT_TIMEOUT = 2 * time.Second
 
+// Error is returned by all functions in this package.
+type Error struct {
+	// Error returned by KT
+	Message string
+	// HTTP status code, if any (0 otherwise)
+	Code int
+}
```
Comment (pavel-odintsov) [inline kt/kt.go]: I also have an idea about function which tries to cast specified arbitrary error type to KT error:
```
func IsKTError(err interface{}) bool {
    _, ok := err.(Error)

    return ok
}
```

And then I could use it this way:
```
if err != nil && kt.IsKTError(err) {
    return "It's KT failure!"
}
```

What do you think about it?

Comment (lmb) [reply kt/kt.go]: Done

**Comment by pavel-odintsov** [comment]
LGTM!

**Pull request merged**